craft_application/app.py

```diff
@@ -16,17 +35,22 @@
     ProvideHelpException,
     emit,
 )
+from craft_providers import Executor, ProviderError
 from xdg import BaseDirectory
 
-from . import errors
-from .lifecycle_commands import get_lifecycle_command_group
+from .commands.base import AppCommand
+from .commands.lifecycle import get_lifecycle_command_group
 from .parts import PartsLifecycle
 from .project import Project
+from .provider import ProviderManager
 
 GLOBAL_VERSION = GlobalArgument(
     "version", "flag", "-V", "--version", "Show the application version and exit"
 )
 
+if TYPE_CHECKING:
```

Comment:
?

Reply:
Leftovers from migrating `constr` etc. to Constrained* types